src/utils.ts

```diff
@@ -167,3 +173,13 @@ export class FormattedMethodDescriptor implements MethodDescriptorProto {
     return result;
   }
 }
+
+export function determineFieldJsonName(field: FieldDescriptorProto, options: Options): string {
+  // By default jsonName is camelCased by the protocol compilier unless the user has
+  // set a "json_name" option on this field. So, if field name is already snake_case
+  // then this will keep jsonName also snake_case.
+  if (field.jsonName.length > 0 && options.snakeToCamel) {
```

Comment:
Hm, we should probably just check `field.jsonName.length` here and skip checking `options.snakeToCamel`? I.e. if there is a `jsonName` we should use it regardless of what the `snakeToCamel` flag is, I think...

Reply:
Hi, I was actually confused with how to handle that case. I also agree with your suggestion regarding `snakeToCamel` flag.